src/src/lms/lms.service.ts

```diff
@@ -419,5 +379,428 @@ export class LMSService {
+				}
+				//issue certificate
+				if (issue_status == 'true') {
+					let certificateTemplate = html_code;
+					let issuance_date = moment().format('YYYY-MM-DD');
+					let issuance_date_tx = moment().format('DD MMM YYYY');
+					let expiration_date = moment(issuance_date)
+						.add(12, 'M')
+						.format('YYYY-MM-DD');
+					const lmsCertificate = new LMSCertificateDto({
+						user_id: user_id,
+						test_id: test_id,
+						certificate_status: 'Issued',
+						issuance_date: issuance_date,
+						expiration_date: expiration_date,
+					});
+					const certificate_data = await this.issueCertificateHtml(
+						lmsCertificate,
+					);
+					if (certificate_data != null) {
+						let certificate_id = certificate_data?.id;
+						let uid = 'P-' + certificate_id + '-' + user_id;
+						//update html code
+						certificateTemplate = certificateTemplate.replace(
+							'{{name}}',
+							user_name,
+						);
+						certificateTemplate = certificateTemplate.replace(
+							'{{issue_date}}',
+							issuance_date_tx,
+						);
+						certificateTemplate = certificateTemplate.replace(
+							'{{user_id}}',
+							uid,
+						);
+
+						//qr code
+						try {
+							let qr_code_verify_link =
+								this.configService.get<string>(
+									'LMS_CERTIFICATE_VERIFY_URL',
+								) +
+								'' +
+								certificate_id;
+
+							let modifiedHtml = null;
+							const modified = await new Promise(
+								(resolve, reject) => {
+									qr.toDataURL(
+										qr_code_verify_link,
+										function (err, code) {
+											if (err) {
+												resolve(null);
+												return;
+											}
+
+											if (code) {
+												const newHtml = code;
+
+												const root =
+													parse(certificateTemplate);
+
+												// Find the img tag with id "qrcode"
+												const qrcodeImg =
+													root.querySelector(
+														'#qr_certificate',
+													);
+
+												if (qrcodeImg) {
+													qrcodeImg.setAttribute(
+														'src',
+														newHtml,
+													);
+													modifiedHtml =
+														root.toString();
+
+													resolve(modifiedHtml);
+												} else {
+													resolve(null);
+												}
+											} else {
+												resolve(null);
+											}
+										},
+									);
+								},
+							);
+							if (modifiedHtml != null) {
+								certificateTemplate = modifiedHtml;
+							}
+						} catch (e) {
+							console.log(e);
+						}
+						//update certificate html
+						const lmsCertificate = new LMSCertificateDto({
+							certificate_html: certificateTemplate,
+						});
+						await this.updateCertificateHtml(
+							lmsCertificate,
+							certificate_id,
+						);
+					} else {
+						//error in create certificate
+					}
+				}
+				// Update in attendance data in database
+				await this.markCertificateStatus(
+					userTestData?.id,
+					issue_status,
+				);
+			}
+		}
+	}
+	async CapitalizeEachWord(sentence) {
```

Comment:
move this to some common helper?

Reply:
created common/method/method.js file and moved this function into it.